client/asset/dcr/dcr.go

```diff
@@ -1976,7 +1976,8 @@ func (dcr *ExchangeWallet) findRedemptionsInTx(scanPoint string, tx *chainjson.T
 		if err != nil {
 			return nil, nil, err
 		}
-		secret, err := dexdcr.FindKeyPush(sigScript, contractHash, dcr.chainParams)
+		// tx.Vin[vin] doesnt have a version field, assume 0
+		secret, err := dexdcr.FindKeyPush(0, sigScript, contractHash, dcr.chainParams)
```

Comment:
The script version pertains to the previous output's pkScript (the p2sh contract output).  We've already validated the contract output at this point, and we only support v0 presently, so we can just hard-code 0 for now.

However, to support multiple script versions, we'd really want to tell `findRedemptionsInTx` the version of each of the outputs we're giving it to check.  In addition, each `findRedemptionReq` could include the script version obtained via `msgTx.TxOut[vout].Version` in `queueFindRedemptionRequest`, and the scan would skip over any outputs with a different script version.

We don't have to do this now, but does that make sense?

Reply:
Makes sense, for the most part.

> the scan would skip over any outputs with an unexpected script version

If the prevOut's script version were available in `tx.Vin[i]`, we can just ignore inputs with script versions different from the script version of the contract output whose redemption is being searched for. Skipping inputs whose sig script returns `ErrUnsupportedScriptVersion` will work fine as well.

Can just do that now.

Reply:
Re the prev out's script version, that was one part of my suggestion, but I think with the `tx.Vin[i].{Vout,Txid}` check, there's no need to pass version in `contractOutpoints` too

Reply:
Oh, no I see what you're saying.  We do need to catch such an error because even now there can be non-0 scripts on the chain.